### PR TITLE
OKAPI-1054: WebClientFactory to avoid socket leaks (okapi-common)

### PR DESCRIPTION
**Pull request by julianladisch**

### CODE REVIEW

File: okapi-common/src/main/java/org/folio/okapi/common/WebClientFactory.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Factory for WebClient instances to avoid web socket leaks.
+ */
+public final class WebClientFactory {
+  private static final Map<Vertx, WebClient> clients = new ConcurrentHashMap<>();
+
+  private WebClientFactory() {
+    throw new UnsupportedOperationException("Utility classes cannot be instantiated");
+  }
```
Comment (adamdickmeiss) [inline okapi-common/src/main/java/org/folio/okapi/common/WebClientFactory.java:16]: Not tested. okapi-testing has a UtilityClassTester.

File: okapi-common/src/main/java/org/folio/okapi/common/WebClientFactory.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  /**
+   * Get a WebClient, returns the same instance for the same Vertx instance.
+   */
+  public static WebClient getWebClient(Vertx vertx) {
+    return clients.computeIfAbsent(vertx, x -> WebClient.create(vertx));
+  }
+}
```
Comment (adamdickmeiss) [inline okapi-common/src/main/java/org/folio/okapi/common/WebClientFactory.java:24]: In some cases, modules pass WebClientOptions to be used. I think they can only be used on construction, so perhaps adding a constructor for that?

File: okapi-common/src/main/java/org/folio/okapi/common/WebClientFactory.java (hunk 1 above)
Comment (julianladisch) [reply okapi-common/src/main/java/org/folio/okapi/common/WebClientFactory.java:17]: Yes, thanks!

File: okapi-common/src/main/java/org/folio/okapi/common/WebClientFactory.java (hunk 2 above)
Comment (julianladisch) [reply okapi-common/src/main/java/org/folio/okapi/common/WebClientFactory.java:37]: Added.

**Review by adamdickmeiss — APPROVED** [review]
Nice

**Pull request merged**